cpp/tests/test_abm_simulation.cpp

```diff
@@ -61,4 +61,28 @@ TEST(TestDiscreteDistribution, generate)
         ASSERT_GE(d, 0);
         ASSERT_LE(d, 4);
     }
+}
+
+TEST(TestSimulation, advance_subpopulation)
+{
+    auto world    = mio::abm::World();
+    auto location = world.add_location(mio::abm::LocationType::School);
```

Comment:
maybe here and in other places in the future, we need to name these *_id to make clear what it returns.

@DavidKerkmann @khoanguyen-dev @xsaschako what do you say?

Reply:
Agreed, in this instance, `school_id` would be a good candidate.

Reply:
Yes, that would add clarity.

Reply:
I changed the location name to have *_id. However, school is a location object so I don't think it should be changed.